src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -35,8 +39,19 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
             EgressArtifactResult result = new();
             try
             {
-                string jsonConfig = Console.ReadLine();
-                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                StdInStream = Console.OpenStandardInput();
+
+                var payloadVersionBuffer = await ReadFromStream(sizeof(int), token); // Payload Version
```

Comment:
I'll let @jander-msft weigh in on this, but a few thoughts:

1. For the sake of not making the manifest overly verbose, we had discussed keeping it more minimal to start with (and expanding it as necessary over time). For fields that weren't initially part of the manifest, from a backwards-compatibility standpoint we would just assume the omission of the field indicates a default (in this case, no version would be interpreted as v1).
2. This might be covered by the startup validation feature I'm working on right now (at least, for extension authors who include the ability to do startup validation). For our extension, when dotnet-monitor starts up we would send a test payload to the extension, and if the version numbers weren't the same (or any other issue) the user would be notified of the failure.

Reply:
I agree with Kyle. The lack of a specified version in the manifest can be interpreted as version 1. Keeping it optional for version 1 protocol with keep things simple in the manifest for now.